src/modules/cfg_rpc/cfg_rpc.c

```diff
@@ -310,48 +310,88 @@ static const char* rpc_get_doc[2] = {
 
 static void rpc_get(rpc_t* rpc, void* c)
 {
-	str	group, var;
-	void	*val;
-	unsigned int	val_type;
-	int	ret;
-	unsigned int	*group_id;
-
-	if (rpc->scan(c, "SS", &group, &var) < 2)
-		return;
-
+	str     group, var;
+	void    *val;
+	unsigned int    val_type;
+	int     ret, n;
+	unsigned int    *group_id;
+
+	n = rpc->scan(c, "SS", &group, &var);
+	/*  2: both group and variable name are present
+	 * -1: only group is present, print all variables in the group */
+	if(n<2) {
+		if (n == -1) {
+			var.s = NULL;
+			var.len = 0;
+		}
+		else return;
+	}
```

Comment:
I think that the right way is to read with optional specifier `*`:

```
n = rpc->scan(c, "S*S", &group, &var);
```

The n is two if both were read or 1 if only group is read.

When n is -1, there can be other errors.

Otherwise, the patch is useful, thanks!

Can you do a new patch based on what I suggested? It can be a follow up of this one, we can squash from the web when merging.

Reply:
Thank you for the review, sure i'll send a new patch with the changes.